### PR TITLE
Feat/rn listener

### PR DESCRIPTION
**Pull request by damien5701**

### CODE REVIEW

File: example/src/App.tsx [hunk 1]
```diff
@@ -1,27 +1,183 @@
...
+    marginBottom: 16,
+  },
+  textStyle: {
+    color: 'white',
+    fontWeight: 'bold',
+    textAlign: 'center',
+  },
+});
+
+// For testing only
```
Comment (imranMnts) [inline example/src/App.tsx]: we can move this part outside of App.tsx

File: src/Components/Main.tsx [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    }
+    return () => {};
+  }, [props.rnRequests, props.reduxActions]);
+
+  const filteredRequests = requests
+    .filter((element) => source === 'ALL' || element.type === source)
+    .filter(
+      (request) =>
+        request.type === 'REDUX' ||
+        (request.type !== 'REDUX' &&
```
Comment (imranMnts) [inline src/Components/Main.tsx]: cannot search a redux action?

File: src/Components/Main.tsx [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+          placeholder={'Search'}
+          onChangeText={onChangeSearch}
+          value={searchQuery}
+        />
+        <IconButton
+          // style={{ marginRight: -8 }}
+          icon={() => <Icon color="white" size={24} name="delete"></Icon>}
+          size={24}
+          onPress={showDialogDelete}
+        ></IconButton>
```
Comment (imranMnts) [inline src/Components/Main.tsx]: cannot use `/>` instead of `></IconButton>`?

File: src/Components/Main.tsx [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+        <IconButton
+          style={{ marginRight: -8 }}
+          icon={() => <Icon color="white" size={24} name="settings"></Icon>}
+          size={24}
+          onPress={showDialog}
+        ></IconButton>
```
Comment (imranMnts) [inline src/Components/Main.tsx]: same

File: src/Components/Main.tsx [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+      </Modal>
+
+      <View style={styles.centeredView}>
+        <View style={styles.modalView}>
+          <FlatList
+            style={styles.flatList}
+            keyExtractor={(item) => `${item._id.toString()}${item.date}`}
+            data={filteredRequests}
+            renderItem={({ item }) =>
+              item.type === 'RNR' || item.type === 'NR' ? (
```
Comment (imranMnts) [inline src/Components/Main.tsx]: we can create an enum to list differents type and give more explicit name

File: src/index.tsx [hunk 6]
```diff
@@ -1,96 +1,62 @@
...
-              >
-                <Text style={styles.textStyle}>Request</Text>
-              </TouchableHighlight>
-            )}
+export const Netwatch: React.FC<IProps> = (props: IProps) => {
+  const [reduxActions, setReduxActions] = useState<Array<ReduxAction>>([]);
+  const [rnRequests, setRnRequests] = useState<Array<RNRequest>>([]);
+  const [showDetails, setShowDetails] = useState(false);
+  const [item, setItem] = useState();
+  _RNLogger.enableXHRInterception();
```
Comment (imranMnts) [inline src/index.tsx]: we should enable it only if user want to turn on Netwatch in his app

Comment (imranMnts) [reply src/index.tsx]: `enabled`

File: src/Components/Main.tsx (hunk 5 above)
Comment (imranMnts) [reply src/Components/Main.tsx]: `LogType` maybe?

File: src/Components/Main.tsx (hunk 2 above)
Comment (damien5701) [reply src/Components/Main.tsx]: It was in WIP. Now it's OK

File: src/Components/Main.tsx (hunk 3 above)
Comment (damien5701) [reply src/Components/Main.tsx]: Done

File: src/Components/Main.tsx (hunk 4 above)
Comment (damien5701) [reply src/Components/Main.tsx]: Done

File: src/index.tsx (hunk 6 above)
Comment (damien5701) [reply src/index.tsx]: Done

File: example/src/App.tsx (hunk 1 above)
Comment (damien5701) [reply example/src/App.tsx]: Done

File: src/Components/Main.tsx (hunk 5 above)
Comment (damien5701) [reply src/Components/Main.tsx]: Done

**Pull request merged**